cmd/main_test.go

```diff
@@ -301,3 +305,72 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
 		"patch",
 		false),
 )
+
+var _ = DescribeTable("TestGetNamespacesWithAccess", func(allNamespaces []k8sapi.Namespace,
+	expectedNs []k8sapi.Namespace, actualNs []k8sapi.Namespace, err error) {
+	e := echo.New()
+	cfg, _ := config.GetConfig()
+	clientset, _ := kubernetes.NewForConfig(cfg)
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	c.Request().Header.Set("X-Email", "test@example.com")
+	authCl := clientset.AuthorizationV1()
+	
+	JustBeforeEach(func() {
+		actualNs, err = getNamespacesWithAccess(e, c, authCl, allNamespaces)
+	})
+
+	Context("When namespaces ns1 and ns2 has all the necessary permissions like create, list, watch and delete", func() {
+		BeforeEach(func() {
+			var ns1, ns2 k8sapi.Namespace
+			ns1, err = createNamespace(k8sClient, "test-tenant")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace ns1: %v", err))
+			ns2, err = createNamespace(k8sClient, "test-tenant-2")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace ns2: %v", err))
+			allNamespaces = []k8sapi.Namespace{ns1, ns2}
+			expectedNs = []k8sapi.Namespace{ns1, ns2}
+		})
+		It("returns all namespaces ns1 and ns2", func() {
+			Expect(actualNs).To(Equal(expectedNs))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
+		})
+	})
+
+	Context("When namspace ns3 doesn't have necessary permissions", func() {
+		BeforeEach(func() {
+			var ns3 k8sapi.Namespace
+			ns3, err = createNamespace(k8sClient, "test-tenant-3")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace ns3: %v", err))
+			allNamespaces = []k8sapi.Namespace{ns3}
+			expectedNs = []k8sapi.Namespace{}
+		})
+		It("doesn't return any namespace", func() {
+			Expect(actualNs).To(Equal(expectedNs))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
+		})
+	})
+
+	Context("When namespaces ns1 and ns2 has all necessary permissions and ns3, ns4 doesn't", func() {
+		BeforeEach(func() {
+			var ns1, ns2, ns3, ns4 k8sapi.Namespace
+			user3 := "user3@konflux.dev"
+			ns1, err = createNamespace(k8sClient, "test-tenant")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace ns1: %v", err))
+			ns2, err = createNamespace(k8sClient, "test-tenant-2")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace ns2: %v", err))
+			ns3, err = createNamespace(k8sClient, "test-tenant-3")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace ns3: %v", err))
+			ns4, err = createNamespace(k8sClient, "test-tenant-4")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace ns4: %v", err))
+			createRole(k8sClient, "test-tenant-4", "namespace-access-3", []string{"create", "list", "watch"})
```

Comment:
Is the point here is giving test-tenant-4 only some of the permissions? it is not entirely clear.